File: SDKs/googleapis.com/vision/v1p2beta1/python/src/sdk/models/shared/googlecloudvisionv1p2beta1word.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import googlecloudvisionv1p2beta1boundingpoly as shared_googlecloudvisionv1p2beta1boundingpoly
from ..shared import googlecloudvisionv1p2beta1textannotationtextproperty as shared_googlecloudvisionv1p2beta1textannotationtextproperty
from ..shared import googlecloudvisionv1p2beta1symbol as shared_googlecloudvisionv1p2beta1symbol


@dataclass_json
@dataclasses.dataclass
class GoogleCloudVisionV1p2beta1Word:
    r"""GoogleCloudVisionV1p2beta1Word
    A word representation.
    """
    
    bounding_box: Optional[shared_googlecloudvisionv1p2beta1boundingpoly.GoogleCloudVisionV1p2beta1BoundingPoly] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('boundingBox') }})
    confidence: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('confidence') }})
    property: Optional[shared_googlecloudvisionv1p2beta1textannotationtextproperty.GoogleCloudVisionV1p2beta1TextAnnotationTextProperty] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('property') }})
    symbols: Optional[list[shared_googlecloudvisionv1p2beta1symbol.GoogleCloudVisionV1p2beta1Symbol]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('symbols') }})
    

File: SDKs/googleapis.com/vision/v1p2beta1/java/lib/src/main/java/openapisdk/models/shared/GoogleCloudVisionV1p2beta1Word.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * GoogleCloudVisionV1p2beta1Word
 * A word representation.
**/
public class GoogleCloudVisionV1p2beta1Word {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("boundingBox")
    public GoogleCloudVisionV1p2beta1BoundingPoly boundingBox;
    public GoogleCloudVisionV1p2beta1Word withBoundingBox(GoogleCloudVisionV1p2beta1BoundingPoly boundingBox) {
        this.boundingBox = boundingBox;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("confidence")
    public Float confidence;
    public GoogleCloudVisionV1p2beta1Word withConfidence(Float confidence) {
        this.confidence = confidence;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("property")
    public GoogleCloudVisionV1p2beta1TextAnnotationTextProperty property;
    public GoogleCloudVisionV1p2beta1Word withProperty(GoogleCloudVisionV1p2beta1TextAnnotationTextProperty property) {
        this.property = property;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("symbols")
    public GoogleCloudVisionV1p2beta1Symbol[] symbols;
    public GoogleCloudVisionV1p2beta1Word withSymbols(GoogleCloudVisionV1p2beta1Symbol[] symbols) {
        this.symbols = symbols;
        return this;
    }
}